aws-datastore/src/test/java/com/amplifyframework/datastore/storage/InMemoryStorageAdapter.java

```diff
@@ -116,23 +116,34 @@ public void initialize(
         queryResultsListener.onResult(result.iterator());
     }
 
+    @Override
+    public <T extends Model> void delete(
+            @NonNull final T item,
+            @NonNull final StorageItemChange.Initiator initiator,
+            @NonNull final ResultListener<StorageItemChange.Record> itemDeletionListener
+    ) {
+        delete(item, initiator, null, itemDeletionListener);
+    }
+
     @SuppressWarnings("unchecked") // item.getClass() -> Class<?>, but type is T. So cast as Class<T> is OK.
     @Override
     public <T extends Model> void delete(
             @NonNull final T item,
             @NonNull final StorageItemChange.Initiator initiator,
+            @Nullable final QueryPredicate predicate,
```

Comment:
`@NonNull`. We have a version of this method that doesn't require a predicate. So, the user should just use that, instead of passing `null` to this version.

Reply:
Feels like unnecessary duplicate code to me. `LocalStorageAdapter` interface already defines predicate to be nullable, so there wouldn't be a point in changing it just for this implementation of the interface.

Reply:
Besides, once predicate evaluator is merged, I will need to add id-check logic on both sides if I do that, which would result in even more duplicate code.

Reply:
Okay, yea. Since `LocalStorageAdapter` isn't user-facing / public, I don't think it's a big deal, either way.